### PR TITLE
Ane 221 include unresolved deps

### PR DESCRIPTION
**Pull request by csasarak**
# Overview

We were excluding any deps without a `resolved` field from npm package-lock.json processing under the assumption that this meant they were workspace packages. This turns out not to be the case, we received an example where regular deps didn't have a resolved location. This PR changes our logic for determining if something is a workspace package to check if that package name matches a workspace in our dependency graph. 

## Acceptance criteria

If this PR is successful, dependency graphs will still include packages that have no `resolved` field for them in `package-lock.json`

## Testing plan

There are unit tests. Additionally there is a test case I can share privately or show if you'd like to test manually with that. 

## Risks

Please review the automated tests and let me know of any cases you think I should include.

## References

[ANE-221](https://fossa.atlassian.net/browse/ANE-221)

## Checklist

- [x] I added tests for this PR's change (or explained in the PR description why tests don't make sense).
- [x] If this PR introduced a user-visible change, I added documentation into `docs/`.
- [x] If this change is externally visible, I updated `Changelog.md`. If this PR did not mark a release, I added my changes into an `# Unreleased` section at the top.
- [x] If I made changes to `.fossa.yml` or `fossa-deps.{json.yml}`, I updated `docs/references/files/*.schema.json`. You may also need to update these if you have added/removed new dependency type (e.g. `pip`) or analysis target type (e.g. `poetry`).

### CODE REVIEW

File: Changelog.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # FOSSA CLI Changelog
 
+## V3.2.11
```
[the remark this answers is not in the document]
Comment (meghfossa) [reply Changelog.md:3]: Can we include Leo's change in this changelog as well? I should have caught this in the PR review. 

```
- Cocoapods - Resolves vendored local podspecs into their source Git repositories when possible. ([#875](https://github.com/fossas/fossa-cli/pull/875))
```

File: src/Strategy/Node.hs [hunk 2]
```diff
@@ -214,6 +216,15 @@ data YarnVersion
   = V1
   | V2Compatible
 
+findWorkspaceNames :: PkgJsonGraph -> WorkspacePackageNames
+findWorkspaceNames PkgJsonGraph{..} =
+  WorkspacePackageNames
+    . Set.fromList
+    $ workspaceNames
+  where
+    childManifests = map snd . AM.edgeList $ jsonGraph
```
Comment (meghfossa) [inline src/Strategy/Node.hs:225]: [optional] [personal nit] types in the where block for readability

File: src/Strategy/Node.hs [hunk 3]
```diff
@@ -214,6 +216,15 @@ data YarnVersion
   = V1
   | V2Compatible
 
+findWorkspaceNames :: PkgJsonGraph -> WorkspacePackageNames
```
Comment (meghfossa) [inline src/Strategy/Node.hs:219]: [optional] [nit] Prefer tiny doc comment explaining how we find these in layman's terms for future code readers :)

**Comment by scruffystuffs** [comment]
Change looks good, still need to comb for nits, but overall this is a nice improvement

File: src/Strategy/Node/Npm/PackageLock.hs [hunk 4]
```diff
@@ -54,7 +53,7 @@ import Effect.Grapher (
 import Effect.ReadFS (ReadFS, readContentsJson)
 import Graphing (Graphing)
 import Path (Abs, File, Path)
-import Strategy.Node.PackageJson (Development, FlatDeps (devDeps, directDeps), NodePackage (pkgName), Production)
+import Strategy.Node.PackageJson (Development, FlatDeps (devDeps, directDeps), NodePackage (pkgName), Production, WorkspacePackageNames (WorkspacePackageNames))
```
Comment (scruffystuffs) [inline src/Strategy/Node/Npm/PackageLock.hs:56]: [nit]: Let's multi-line this.

File: test/Node/NodeSpec.hs [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+import Data.Glob (unsafeGlobRel)
+import Data.Map qualified as Map
+import Graphing qualified
+import Path (Abs, Dir, Path, mkRelDir, mkRelFile, (</>))
+import Path.IO (getCurrentDir)
+import Strategy.Node (NodeProject (NPMLock), discover, getDeps)
+import Strategy.Node.PackageJson (Manifest (..), PackageJson (PackageJson, packageDeps, packageDevDeps, packageLicense, packageLicenses, packageName, packagePeerDeps, packageVersion, packageWorkspaces), PkgJsonGraph (PkgJsonGraph, jsonGraph, jsonLookup), PkgJsonLicense (LicenseText), PkgJsonWorkspaces (PkgJsonWorkspaces, unWorkspaces))
+import Test.Effect (it', shouldBe')
+import Test.Hspec (Spec, describe, runIO)
+import Types (DependencyResults (DependencyResults, dependencyGraph, dependencyGraphBreadth, dependencyManifestFiles), DiscoveredProject (DiscoveredProject, projectBuildTargets, projectData, projectPath, projectType), DiscoveredProjectType (NpmProjectType), FoundTargets (ProjectWithoutTargets), GraphBreadth (Complete))
```
Comment (scruffystuffs) [inline test/Node/NodeSpec.hs:17]: [nit]: Let's multi-line this.

File: test/Node/NodeSpec.hs [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+
+import Algebra.Graph.AdjacencyMap qualified as AM
+import Data.Foldable (for_)
+import Data.Glob (unsafeGlobRel)
+import Data.Map qualified as Map
+import Graphing qualified
+import Path (Abs, Dir, Path, mkRelDir, mkRelFile, (</>))
+import Path.IO (getCurrentDir)
+import Strategy.Node (NodeProject (NPMLock), discover, getDeps)
+import Strategy.Node.PackageJson (Manifest (..), PackageJson (PackageJson, packageDeps, packageDevDeps, packageLicense, packageLicenses, packageName, packagePeerDeps, packageVersion, packageWorkspaces), PkgJsonGraph (PkgJsonGraph, jsonGraph, jsonLookup), PkgJsonLicense (LicenseText), PkgJsonWorkspaces (PkgJsonWorkspaces, unWorkspaces))
```
Comment (scruffystuffs) [inline test/Node/NodeSpec.hs:14]: [nit]: Let's multi-line this.

File: test/Node/NodeSpec.hs [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+  pkgJsonWorkspaceSpec currDir
+  npmLockAnalysisSpec currDir
+
+discoveredWorkSpaceProj :: Path Abs Dir -> DiscoveredProject NodeProject
+discoveredWorkSpaceProj currDir =
+  DiscoveredProject
+    { projectType = NpmProjectType
+    , projectPath = currDir </> $(mkRelDir "test/Node/testdata/workspace-test/")
+    , projectBuildTargets = ProjectWithoutTargets
+    , projectData =
```
Comment (scruffystuffs) [inline test/Node/NodeSpec.hs:31]: [no action]: I don't love this massive literal data block, but I don't really have a good alternative.

**Review by scruffystuffs — APPROVED** [review]
Just indentation nits.

Also, good call on testing node discovery, since it's not a trivial discovery function.  🔥

File: test/Node/NodeSpec.hs (hunk 7 above)
Comment (csasarak) [reply test/Node/NodeSpec.hs:58]: It's not my favorite and I can't think of an alternative either. I've reduced a bit of the vertical space it takes up, but that doesn't change much fundamentally.
